src/js/controllers/ctrl-widget-settings.js

```diff
@@ -50,35 +50,46 @@ app.controller('WidgetSettingsController', function(
 	}
 
 	const _toggleNormalAccess = () => {
-		if (($scope.guestAccess = true)) {
-			$scope.guestAccess = false
-		}
-		if (($scope.embeddedOnly = true)) {
+		// disallow student made widgets from being toggled out of guest mode
+		if ($scope.studentMade) return
+
+		$scope.guestAccess = !$scope.guestAccess
+
+		if ($scope.embeddedOnly == true) {
 			$scope.embeddedOnly = false
 		}
+
+		// warn user that students will be removed if guest mode is disabled
+		if ($scope.selected.widget.student_access == true && $scope.guestAccess == false) {
+			$scope.alert.msg =
+				'Warning: Disabling Guest Mode will automatically revoke access to this widget for any students it has been shared with!'
+			$scope.alert.title = 'Students with access will be removed'
+			$scope.alert.fatal = false
+		}
+
+		_updateSliderAfterChange()
```

Comment:
This should get the slider updating to the right state. I moved the delay that was in one of the functions into it's own method and just call it on all 3 methods